File: frontend/src/types/project.ts
export interface Project {
  id: string;
  company_id: string;
  name: string;
  description: string | null;
  default_product_id: string | null;
  jurisdiction: string | null;
  target_payers: string[];
  analysis_scope: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreate {
  company_id: string;
  name: string;
  description?: string;
  jurisdiction?: string;
}

export interface Product {
  id: string;
  company_id: string;
  name: string;
  description: string | null;
  product_type: string | null;
  regulatory_stage: string | null;
  fda_status: string | null;
  intended_use: string | null;
  indications_for_use: string | null;
  target_population: string | null;
  intended_user: string | null;
  site_of_service: string | null;
  care_setting: string | null;
  clinical_output: string | null;
  ai_role: string | null;
  hardware_version: string | null;
  software_version: string | null;
  model_version: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProductCreate {
  name: string;
  description?: string;
  product_type?: string;
  regulatory_stage?: string;
  fda_status?: string;
  intended_use?: string;
}

File: backend/app/schemas/project.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    company_id: uuid.UUID
    name: str
    description: str | None = None
    default_product_id: uuid.UUID | None = None
    jurisdiction: str | None = "United States"
    target_payers: list[str] = []
    analysis_scope: str | None = None


class ProjectUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    default_product_id: uuid.UUID | None = None
    jurisdiction: str | None = None
    target_payers: list[str] | None = None
    analysis_scope: str | None = None


class ProjectRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    company_id: uuid.UUID
    name: str
    description: str | None
    default_product_id: uuid.UUID | None
    jurisdiction: str | None
    target_payers: list[str]
    analysis_scope: str | None
    created_at: datetime
    updated_at: datetime
